diff --git a/Makefile.nanvix b/Makefile.nanvix
index 0f9703e9a2..66c67c2814 100644
--- a/Makefile.nanvix
+++ b/Makefile.nanvix
@@ -76,9 +76,9 @@ $(CROSS_FILE):
 	@echo '' >> $@
 	@echo '[built-in options]' >> $@
 	@echo "c_args = ['-I$(SYSROOT_PATH)/include', '-I$(SYSROOT_PATH)/include/python3.12']" >> $@
-	@echo "c_link_args = ['-static', '-L$(SYSROOT_PATH)/lib', '-T$(SYSROOT_PATH)/lib/user.ld', '-Wl,--start-group', '$(SYSROOT_PATH)/lib/libposix.a', '$(TOOLCHAIN_PREFIX)/i686-nanvix/lib/libc.a', '$(TOOLCHAIN_PREFIX)/i686-nanvix/lib/libm.a', '-Wl,--end-group']" >> $@
+	@echo "c_link_args = ['-static', '-L$(SYSROOT_PATH)/lib', '-T$(SYSROOT_PATH)/lib/user.ld', '-Wl,--allow-multiple-definition', '-Wl,--start-group', '$(SYSROOT_PATH)/lib/libposix.a', '$(TOOLCHAIN_PREFIX)/i686-nanvix/lib/libc.a', '$(TOOLCHAIN_PREFIX)/i686-nanvix/lib/libm.a', '-Wl,--end-group']" >> $@
 	@echo "cpp_args = ['-I$(SYSROOT_PATH)/include', '-I$(SYSROOT_PATH)/include/python3.12']" >> $@
-	@echo "cpp_link_args = ['-static', '-L$(SYSROOT_PATH)/lib', '-T$(SYSROOT_PATH)/lib/user.ld', '-Wl,--start-group', '$(SYSROOT_PATH)/lib/libposix.a', '$(TOOLCHAIN_PREFIX)/i686-nanvix/lib/libc.a', '$(TOOLCHAIN_PREFIX)/i686-nanvix/lib/libm.a', '-lstdc++', '-Wl,--end-group']" >> $@
+	@echo "cpp_link_args = ['-static', '-L$(SYSROOT_PATH)/lib', '-T$(SYSROOT_PATH)/lib/user.ld', '-Wl,--allow-multiple-definition', '-Wl,--start-group', '$(SYSROOT_PATH)/lib/libposix.a', '$(TOOLCHAIN_PREFIX)/i686-nanvix/lib/libc.a', '$(TOOLCHAIN_PREFIX)/i686-nanvix/lib/libm.a', '-lstdc++', '-Wl,--end-group']" >> $@
 	@echo '' >> $@
 	@echo '[host_machine]' >> $@
 	@echo "system = 'nanvix'" >> $@
@@ -95,7 +95,7 @@ configure: $(CONFIGURED_MARKER)
 
 $(CONFIGURED_MARKER): $(CROSS_FILE)
 	PKG_CONFIG_LIBDIR="$(SYSROOT_PATH)/lib/pkgconfig" \
-	/usr/bin/python3 vendored-meson/meson/meson.py setup $(BUILDDIR) \
+	python3 vendored-meson/meson/meson.py setup $(BUILDDIR) \
 	    --cross-file $(CROSS_FILE) \
 	    --pkg-config-path="$(SYSROOT_PATH)/lib/pkgconfig" \
 	    -Dallow-noblas=true \
